virtualizarr/manifests/group.py

```diff
@@ -131,3 +135,70 @@ def to_virtual_dataset(self) -> xr.Dataset:
             coord_names=coord_names,
             attrs=attributes,
         )
+
+    @classmethod
+    def from_virtual_dataset(
+        cls,
+        vds: xr.Dataset,
+    ) -> "ManifestGroup":
+        """
+        Create a new ManifestGroup from a virtual dataset object.
+
+        The virtual dataset should contain only virtual variables, i.e. those backed by ManifestArrays.
+
+        Parameters
+        ----------
+        vds: xr.Dataset
+            Virtual dataset, containing only virtual variables.
+        """
+
+        for name, var in vds.variables.items():
+            if not isinstance(var.data, ManifestArray):
+                raise TypeError(
+                    f"Cannot convert a dataset containing a loadable variable directly to a ManifestGroup, but found variable {name} has type {type(var.data)}"
+                )
+
+        manifestarrays = {name: var.data for name, var in vds.variables.items()}
+
+        attributes = vds.attrs
+        # TODO test this is correct
+        attributes["dimension_names"] = " ".join(list(vds.dims))
+        # TODO test this constructor round-trips coordinates
+        attributes["coordinates"] = " ".join(list(vds.coords))
+
+        return cls(arrays=manifestarrays, attributes=attributes)
+
+    def to_icechunk(
```

Comment:
The point of adding this on `ManifestGroup` instead of `ManifestStore` initially is to make it easier to compose to deal with virtual datatrees later.